source/_posts/angular-session-storage.md

```diff
@@ -0,0 +1,299 @@
+```
+
+{% img "save-data.jpg" "Save data in angular session storage" "lazy" %}
+
+When you click the "Save data to sessionStorage" button, the key "name" and the value "Sanjeev" will be saved.
+
+This can be verified by inspecting the page and going to `application > session storage`.
+
+{% img "session-storage.jpg" "Verify saved data in angular session storage" "lazy" %}
+
+If we close the browser or this tab, the session data stored will be automatically cleared.
+
+### Retrieve data from session storage
+
+We will add a `get()` function to our `app.component.ts` file to retrieve and display the data stored in session storage.
+
+```typescript
+get(){
+}
+```
+Within the `get()` function, we will use `getItem` to retrieve data from the `sessionStorage` based on the key.
+
+Syntax:
+
+```typescript
+sessionStorage.getItem('key');
+```
+
+Thus, this is how our code in the `app.component.ts` will appear:
+
+```typescript
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.css']
+})
+export class AppComponent {
+  title = 'new_project';
+
+  dataSave(){
+    sessionStorage.setItem('name', 'Sanjeev');
+  }
+
+  get(){
+    return sessionStorage.getItem('name');
+  }
+}
+```
+
+To display the data stored in session storage, add a `p` tag to the `app.component.html` file and call the `get` method within that `p` tag.
+
+```html
+<h1>Angular sessionStorage</h1>
+<button (click)="dataSave()">Save data to sessionStorage</button>
+<p>{{ get() }}</p>
+```
+
+{% img "view-data.jpg" "View Data in angular session storage" "lazy" %}
+
+### Deleting data based on key
+
+For deleting the data saved in session storage based on the key, first, we will create a `dataRemove()` function in the `app.component.ts` file.
+
+```typescript
+dataRemove(){
+}
+```
+
+Inside the `dataRemove()` function add `removeItem` to remove specific data from the `sessionStorage` based on key.
+
+Syntax:
+
+```typescript
+localStorage.removeItem("key");
+```
+
+Thus, this is how `app.component.ts` will appear:
+
+```typescript
+dataRemove(){
+  sessionStorage.removeItem('name');
+}
+```
+
+After this, add a button in `app.component.html` file that'll help us remove the stored data.
+
+```html
+<button (click)="dataRemove()">Remove data from sessionStorage</button>
+```
+
+{% img "remove-data.jpg" "Remove data in angular session storage" "lazy" %}
+
+> Note: To clarify things, you can store multiple data in session storage and then use this method to remove specific data.
+
+For example, I've added a few additional data to our session storage, and I'll specifically remove the data with key value as Name.
+
+```typescript
+dataSave(){
+    sessionStorage.setItem('Name', 'Sanjeev');
+    sessionStorage.setItem('Country', 'India');
+    sessionStorage.setItem('Age', '22');
+  }
+dataRemove(){
+    sessionStorage.removeItem('Name');
+```
+
+### Delete all data from session storage

```

Comment:
It must be consistent. If you decided to use uppercase in headlines you should do it here as well. We talked about it earlier. It should be:
`Delete all data from Session Storage`